awscli/customizations/s3/utils.py

```diff
@@ -493,22 +493,23 @@ def _decode_keys(self, parsed, **kwargs):
 class ScopedEventHandler(object):
     """Register an event callback for the duration of a scope."""
 
-    def __init__(self, session, event_name, handler, unique_id=None,
+    def __init__(self, event_emitter, event_name, handler, unique_id=None,
                  unique_id_uses_count=False):
-        self._session = session
+        self._event_emitter = event_emitter
         self._event_name = event_name
         self._handler = handler
         self._unique_id = unique_id
         self._unique_id_uses_count = unique_id_uses_count
 
     def __enter__(self):
-        self._session.register(self._event_name, self._handler, self._unique_id,
-                               self._unique_id_uses_count)
+        self._event_emitter.register(self._event_name, self._handler,
+                                     self._unique_id,
+                                     self._unique_id_uses_count)
```

Comment:
The test is not invalid because it was erroring on a ``test_plugin.py`` integration test...

I can dive a little bit deeper into this to see if it is an issue with botocore.